### PR TITLE
December release highlights for JavaScript & TypeScript

### PR DESCRIPTION
**Pull request by sadasant**
December release notes for JavaScript.

Please suggest changes with your release notes.

If you're looking for inspiration:

- Highlight guidelines: https://azure.github.io/azure-sdk/policies_releasenotes.html
- September's PR: https://github.com/Azure/azure-sdk/pull/1750
- October's PR: https://github.com/Azure/azure-sdk/pull/1878
- November's PR: https://github.com/Azure/azure-sdk/pull/2029

### CODE REVIEW

File: releases/2020-12/js.md [hunk 1]
```diff
@@ -34,9 +43,36 @@ If you have a bug or feature request for one of the libraries, please post an is
...
+
+=== COPY THIS AND ADD THE INFORMATION OF YOUR PACKAGE: ===
+
+Keep in mind that:
+
+- Including the package name in the headers makes the URL links work for multiple packages.
+- The format of this file will be cleaned up once all of your proposals are in.
+
+---
+
```
Comment (jonathandturner) [inline releases/2020-12/js.md:56]: ### Synapse

This is the first release of the Synapse packages to support the new Azure Synapse service. For more information about the Synapse service, please [visit the Microsoft Synapse homepage](https://azure.microsoft.com/en-us/services/synapse-analytics/).

File: releases/2020-12/js.md [hunk 2]
```diff
@@ -8,24 +8,33 @@ repository: azure/azure-sdk-for-js
...
 To install the packages, copy and paste the below into a terminal.
 
 ```bash
-$> npm install @azure/package-name
+$> npm install @azure/synapse-accesscontrol
+$> npm install @azure/synapse-artifacts
+$> npm install @azure/synapse-spark
+$> npm install @azure/synapse-managed-private-endpoints
+$> npm install @azure/synapse-monitoring
+$> npm install @azure/storage-blob
```
Comment (ljian3377) [inline releases/2020-12/js.md:34]:
```suggestion
$> npm install @azure/storage-blob@next
```

File: releases/2020-12/js.md [hunk 3]
```diff
@@ -8,24 +8,33 @@ repository: azure/azure-sdk-for-js
...
+$> npm install @azure/storage-file-share
```
Comment (ljian3377) [inline releases/2020-12/js.md:35]:
```suggestion
$> npm install @azure/storage-file-share@next
```

File: releases/2020-12/js.md [hunk 4]
```diff
@@ -8,24 +8,33 @@ repository: azure/azure-sdk-for-js
...
+$> npm install @azure/storage-file-datalake
```
Comment (ljian3377) [inline releases/2020-12/js.md:36]:
```suggestion
$> npm install @azure/storage-file-datalake@next
```

File: releases/2020-12/js.md [hunk 5]
```diff
@@ -8,24 +8,33 @@ repository: azure/azure-sdk-for-js
...
+$> npm install @azure/storage-queue
```
Comment (ljian3377) [inline releases/2020-12/js.md:37]:
```suggestion
$> npm install @azure/storage-queue@next
```

File: releases/2020-12/js.md [hunk 6]
```diff
@@ -34,9 +43,36 @@ If you have a bug or feature request for one of the libraries, please post an is
...
-- Major changes only!
+##### New Features on @azure/_package-name_@_version_
+
+- _Add one or more, or remove the "New Features on ..." section._
+
+##### Major Fixes on @azure/_package-name_@_version_
+
+- _Add one or more, or remove the "Major Fixes on ..." section._
+
+---
```
Comment (ljian3377) [inline releases/2020-12/js.md:76]:
```suggestion

### Azure Storage

#### @azure/storage-blob [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/CHANGELOG.md)

We are releasing to add support for new service features in Azure Storage Service API version 2020-04-08.

##### New Features on @azure/storage-blob@12.4.0-beta.1

- Added a new interface `BlockBlobClient.syncUploadFromURL()` to support creating a new Block Blob where the contents of the blob are read from a given URL.
- `BlobClient.setTags()` and `BlobClient.getTags()` now support the `LeaseAccessConditions` and `BlobServiceClient.findBlobsByTags()` will return all matching tags for each blob.
- Added `generateSasUrl` to `BlobClient` and `ContainerClient` to generate a service-level SAS URI for the client.
- Added `generateAccountSasUrl` to `BlobServiceClient` to generate an account-level SAS URI for the client.

#### @azure/storage-file-datalake [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-datalake/CHANGELOG.md)

##### New Features on @azure/storage-blob@12.3.0-beta.1

- Added `generateSasUrl` to `DataLakeFileSystemClient`, `DataLakeDirectoryClient` and `DataLakeFileClient` to generate a service-level SAS URI for the client.
- Added `generateAccountSasUrl` to `DataLakeServiceClient` to generate an account-level SAS URI for the client.

#### @azure/storage-file-share [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/CHANGELOG.md)

##### New Features on @azure/storage-file-share@12.4.0-beta.1

- Share now supports for NFS. User can specify the `protocols` when creating a Share via `ShareClient.create()`. Also added an `rootSquash` option for NFS shares.
- Added `generateSasUrl` to `ShareClient` and `ShareFileClient` to generate a service-level SAS URI for the client.
- Added `generateAccountSasUrl` to `ShareServiceClient` to generate an account-level SAS URI for the client.

#### storage-queue [Changelog](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-queue/CHANGELOG.md)

##### New Features on @azure/storage-queue@12.3.0-beta.1

- Added `generateSasUrl` to `QueueClient` to generate a service-level SAS URI for the client.
- Added `generateAccountSasUrl` to `QueueServiceClient` to generate an account-level SAS URI for the client.

```

File: releases/2020-12/js.md (hunk 1 above)
Comment (sadasant) [reply releases/2020-12/js.md:59]: I've added something very close to this over here: https://github.com/Azure/azure-sdk/pull/2138/files#diff-a0dc4e4fee6b1e962f28dfe909e1694f6f1b1b3b2dcc8c8cb33278a66c51a7ddR46 please review!

File: releases/2020-12/js.md (hunk 6 above)
Comment (sadasant) [reply releases/2020-12/js.md:91]: I have added this and your other suggestions to this branch! Thank you.

File: releases/2020-12/js.md [hunk 7]
```diff
@@ -8,24 +8,33 @@ repository: azure/azure-sdk-for-js
...
-
-- _Add packages_
-
-#### Updates
-
-- _Add packages_
-
 #### Beta
 
 - _Add packages_
```
Comment (ramya-rao-a) [inline releases/2020-12/js.md:13]:
```suggestion
```

File: releases/2020-12/js.md [hunk 8]
```diff
@@ -34,9 +43,51 @@ If you have a bug or feature request for one of the libraries, please post an is
 
 ## Release highlights
 
-### _Package name_
+### Synapse
+
+This is the first release of the Synapse packages to support the new Azure Synapse service. For more information about the Synapse service, please [visit the Microsoft Synapse homepage](https://azure.microsoft.com/en-us/services/synapse-analytics/).
```
Comment (ramya-rao-a) [inline releases/2020-12/js.md:48]:
```suggestion
This is the first release of the Synapse packages to support the new [Azure Synapse Analytics service](https://azure.microsoft.com/en-us/services/synapse-analytics/).
```

File: releases/2020-12/js.md [hunk 9]
```diff
@@ -34,9 +43,51 @@ If you have a bug or feature request for one of the libraries, please post an is
...
+
+The packages released under the Synapse name are as follow:
+
+- <https://www.npmjs.com/package/@azure/synapse-access-control/v/1.0.0-beta.1>
+- <https://www.npmjs.com/package/@azure/synapse-artifacts/v/1.0.0-beta.1>
+- <https://www.npmjs.com/package/@azure/synapse-spark/v/1.0.0-beta.1>
+- <https://www.npmjs.com/package/@azure/synapse-managed-private-endpoints/v/1.0.0-beta.1>
+- <https://www.npmjs.com/package/@azure/synapse-monitoring/v/1.0.0-beta.1>
```
Comment (ramya-rao-a) [inline releases/2020-12/js.md:56]: Any reason to use the complete urls like this instead of using the package name as the link text?

Comment (sadasant) [reply releases/2020-12/js.md:56]: Not really! I'll switch to package name (with link).

File: releases/2020-12/js.md [hunk 10]
```diff
@@ -34,9 +42,51 @@ If you have a bug or feature request for one of the libraries, please post an is
...
+
+This is the first release of the Synapse packages to support the new [Azure Synapse Analytics service](https://azure.microsoft.com/en-us/services/synapse-analytics/).
+
+The packages released under the Synapse name are as follow:
+
+- [Azure Synapse Access Control](https://www.npmjs.com/package/@azure/synapse-access-control/v/1.0.0-beta.1)
+- [Azure Synapse Artifacts](https://www.npmjs.com/package/@azure/synapse-artifacts/v/1.0.0-beta.1)
+- [Azure Synapse Spark](https://www.npmjs.com/package/@azure/synapse-spark/v/1.0.0-beta.1)
+- [Azure Synapse Managed Private Endpoints](https://www.npmjs.com/package/@azure/synapse-managed-private-endpoints/v/1.0.0-beta.1)
+- [Azure Synapse Monitoring](https://www.npmjs.com/package/@azure/synapse-monitoring/v/1.0.0-beta.1)
```
Comment (ramya-rao-a) [inline releases/2020-12/js.md:55]: I think the name of the package would be useful. How about using this format?

`Azure Synapse Monitoring: [@azure/synapse-monitoring](https://www.npmjs.com/package/@azure/synapse-monitoring)`

Comment (sadasant) [reply releases/2020-12/js.md:55]: can do!

**Review by czubair — APPROVED** [review]

**Review by ramya-rao-a — APPROVED** [review]

**Pull request merged**